server/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -96,7 +99,19 @@ public IngestService(ClusterService clusterService, ThreadPool threadPool,
                 threadPool.getThreadContext(), threadPool::relativeTimeInMillis,
                 (delay, command) -> threadPool.schedule(
                     command, TimeValue.timeValueMillis(delay), ThreadPool.Names.GENERIC
-                ), this
+                ), this, index -> {
+                    IndexService indexService = indicesService.indexService(index);
+                    if (indexService == null) {
+                        throw new ResourceNotFoundException("index [" + index + "] not locally allocated");
+                    }
+                    int numShards = indexService.getMetaData().getNumberOfShards();
+                    if (numShards != 1) {
+                        throw new IllegalStateException("index [" + index + "] must have 1 shard, but has " + numShards + " shards");
+                    }
+
+                    IndexShard indexShard = indexService.getShard(0);
+                    return indexShard.acquireSearcher("ingest");
```

Comment:
for my education, does this create a new searcher or give you back a pre-existing one ? If the former, do we need to do anything with the existing one w/r/t disabling refreshes ? (sorry if this a bogus question, I don't have much experience at this level).

Reply:
This always creates a new searcher. The code invoking this should be aware of this and make sure to close it after it is done with the searcher. In our case we will need to reuse the searcher from an ingest processor. We don't want to invoke this every time an enrich processor is invoked.